tests/vm_test.go

```diff
@@ -714,519 +699,411 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
-						return vmi.Status.Phase == v1.Running
-					}, 240*time.Second, 1*time.Second).Should(BeTrue())
-				})
-
-				It("[test_id:2035] should restart", func() {
-					By("creating a VM with RunStrategyManual")
-					vm := newVirtualMachineWithRunStrategy(v1.RunStrategyManual)
-
-					startCommand := clientcmd.NewRepeatableVirtctlCommand(virtctl.COMMAND_START, "--namespace", vm.Namespace, vm.Name)
-					stopCommand := clientcmd.NewRepeatableVirtctlCommand(virtctl.COMMAND_STOP, "--namespace", vm.Namespace, vm.Name)
-					restartCommand := clientcmd.NewRepeatableVirtctlCommand(virtctl.COMMAND_RESTART, "--namespace", vm.Namespace, vm.Name)
+			It("should restart a failed VMI", func() {
+				By("Creating a VM with RunStrategyRerunOnFailure")
+				vm := libvmi.NewVirtualMachine(libvmifact.NewGuestless(), libvmi.WithRunStrategy(v1.RunStrategyRerunOnFailure))
+				vm, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Create(context.Background(), vm, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
 
-					By("Invoking virtctl restart should fail")
-					Expect(restartCommand()).ToNot(Succeed())
+				By("Waiting for VM to exist")
+				Eventually(ThisVM(vm), 360*time.Second, 1*time.Second).Should(BeReady())
 
-					By("Invoking virtctl start")
-					Expect(startCommand()).To(Succeed())
+				By("Waiting for VM to start")
+				Eventually(ThisVM(vm), 360*time.Second, 1*time.Second).Should(HavePrintableStatus(v1.VirtualMachineStatusRunning))
 
-					By("Waiting for VM to be ready")
-					Eventually(ThisVM(vm), 240*time.Second, 1*time.Second).Should(BeReady())
+				vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
 
-					By("Invoking virtctl stop")
-					Expect(stopCommand()).To(Succeed())
+				By("Triggering a segfault in qemu")
+				domSpec, err := tests.GetRunningVMIDomainSpec(vmi)
+				Expect(err).ToNot(HaveOccurred())
+				emulator := filepath.Base(domSpec.Devices.Emulator)
+				libpod.RunCommandOnVmiPod(vmi, []string{"killall", "-11", emulator})
 
-					By("Ensuring the VirtualMachineInstance is stopped")
-					Eventually(ThisVM(vm), 240*time.Second, 1*time.Second).Should(Not(BeCreated()))
+				By("Ensuring the VM stops")
+				Eventually(ThisVM(vm), 360*time.Second, 1*time.Second).Should(HavePrintableStatus(v1.VirtualMachineStatusStopped))
 
-					By("Waiting state change request to clear for stopped VM")
-					Eventually(ThisVM(vm), 240*time.Second, 1*time.Second).Should(Not(HaveStateChangeRequests()))
+				By("Waiting for VM to start again")
+				Eventually(ThisVM(vm), 360*time.Second, 1*time.Second).Should(HavePrintableStatus(v1.VirtualMachineStatusRunning))
+			})
+		})
 
-					By("Invoking virtctl start")
-					Expect(startCommand()).To(Succeed())
+		It("Should leave a failed VMI using RunStrategyOnce", Serial, func() {
```

Comment:
```suggestion
		It("Should leave a failed VMI using RunStrategyOnce", func() {
```
nit: remove the Serial decorator

Reply:
Not at all a nit, thanks for having an eye on it. I missed that, also managed to put that change into the wrong commit initially.

Reply:
Glad to help :)